github/Requester.py

```diff
@@ -314,7 +314,8 @@ def __init__(
         self.__app_auth = app_auth
         self.__base_url = base_url
 
-        self.__auth_lock = RLock()
+        if self.__app_auth is not None:
+            self.__auth_lock = RLock()
```

Comment:
What about:

```suggestion
            try:
                self.__auth_lock = RLock()
            except OSError:
                from contextlib import contextmanager

                @contextmanager
                def noop_lock():
                    yield

                self.__auth_lock = noop_lock()
```

Reply:
We have two choices and they're both terrible -- attempting to lock if using app auth and failing on Lambda, or having no locking at all on Lambda. They are both better than the current situation of just failing no matter waht.